tests/testrides.py

```diff
@@ -0,0 +1,121 @@
+            "destination": "Ngara",
+            "route": "Thika superhighway",
+            "start time": "111111",
+            "available space": 10
+        }
+        response = self.app.post('/api/v1/rides',
+                                 data=json.dumps(ride),
+                                 content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'],
+                         "use correct format for date and time.")
+
+    def test_cannot_create_ride_without_details(self):
+        ride = {}
+        response = self.app.post('/api/v1/rides',
+                                 data=json.dumps(ride),
+                                 content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'],
+                         "make sure you provide all required fields.")
+
+    def test_available_space_can_only_be_numbers(self):
+        ride = {
+            "start point": "Witeithye",
+            "destination": "Ngara",
+            "route": "Thika superhighway",
+            "start time": "June 10 2018 6:00AM",
+            "available space": "10"
+        }
+        response = self.app.post('/api/v1/rides',
+                                 data=json.dumps(ride),
+                                 content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'],
+                         "available space can only be numbers.")
+
+    def test_user_can_get_all_rides(self):
+        """test user can get available ride offers"""
+        response = self.app.get('/api/v1/rides',
+                                content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+
+    def test_user_can_get_a_ride(self):
+        """test user can get a single ride offer"""
+        response = self.app.get('/api/v1/rides/1',
+                                content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['id'], 1)
+
+    def test_user_cannot_get_ride_that_does_not_exist(self):
+        """Assumes no ride with a negative id number"""
+        response = self.app.get('/api/v1/rides/-1',
+                                content_type='application/json')
+        self.assertEqual(response.status_code, 404)
+        response_data = json.loads(response.get_data().decode('utf-8'))
+        self.assertEqual(response_data['message'], 'Ride does not exist')
+
+    def test_user_can_join_a_ride(self):
+        """test user can join a ride"""
```

Comment:
I assume making a request is not actually joining the ride. My suggestion therefore, is to rename 
`test_user_can_join_a_ride` to `test_user_can_request_a_ride` and 
`test_user_cannot_join_a_non_existing_ride` to `test_user_cannot_request_a_non_existing_ride`

Reply:
This has been taken care of